radixdlt-core/radixdlt/src/integration/java/com/radixdlt/harness/simulation/monitors/EventNeverOccursInvariant.java

```diff
@@ -76,21 +77,27 @@
 public final class EventNeverOccursInvariant<T> implements TestInvariant {
   private final NodeEvents nodeEvents;
   private final Class<T> eventClass;
+  private final Predicate<T> predicate;
 
-  public EventNeverOccursInvariant(NodeEvents nodeEvents, Class<T> eventClass) {
+  public EventNeverOccursInvariant(
+      NodeEvents nodeEvents, Class<T> eventClass, Predicate<T> predicate) {
     this.nodeEvents = nodeEvents;
     this.eventClass = eventClass;
+    this.predicate = predicate;
   }
 
   @Override
   public Observable<TestInvariantError> check(SimulationNodes.RunningNetwork network) {
     return Observable.<TestInvariant.TestInvariantError>create(
             emitter ->
                 this.nodeEvents.addListener(
-                    (node, event) ->
+                    (node, event) -> {
+                      if (predicate.test(event)) {
                         emitter.onNext(
                             new TestInvariant.TestInvariantError(
-                                "Event " + event + " occurred at node " + node)),
+                                "Event " + event + " occurred at node " + node));
+                      }
+                    },
```

Comment:
Perhaps it worth to extract lambda into dedicated method to reduce nesting level and separate abstractions.